lib/nanoid.rb

```diff
@@ -14,8 +14,8 @@ def self.generate(size: 21, alphabet: SAFE_ALPHABET)
   def self.simple_generate(size:)
     bytes = random_bytes(size)
 
-    (0...size).reduce('') do |acc, i|
-      acc + SAFE_ALPHABET[bytes[i] & 63]
+    (0...size).reduce(''.dup) do |acc, i|
```

Comment:
Why duplicate empty string?

Reply:
All your strings are frozen now globally. That's why you will catch error after `<<` mutation. `.dup` creates unfrozen string.

Reply:
I see. I forgot i freeze it.
Then do you have any speed test that your solution is more performant?

Reply:
I run one of your unit tests and checked objects allocation with https://github.com/srawlins/allocation_stats
For 10000 runs it shows 2 times less Strings allocations (190K vs 100K) and about 1 second less time execution.

Of course it is micro-optimization but `micro` is more that `nano` :)

Reply:
Sure, every byte of memory counts :)